### PR TITLE
Fixing nrnivcore-modl in build (& cleanup) +CI

### PR DESCRIPTION
**Pull request by ferdonline**
This PR fixes a few problems with nrnivmodl-core

- The **build tree** was not complete, so it couldnt find some headers and the nrnunits.lib while calling nrnivmodl-core from the CoreNeuron build directory
- Loading stored settings was having precedence of CLI arguments

We now loop over all the headers and `configure_file` on them so they are copied over on changes

### CODE REVIEW

File: coreneuron/CMakeLists.txt [hunk 1]
```diff
@@ -205,19 +205,19 @@ set(CORENRN_LINK_LIBS "${CORENRN_LINK_LIBS}" PARENT_SCOPE)
...
-configure_file("nrnoc/nrnoc_ml.ispc" ${CMAKE_BINARY_DIR}/include/coreneuron/nrnoc/nrnoc_ml.ispc)
-
-# TODO : this doesn't get re-copied if source changes
-file(COPY "utils/randoms" DESTINATION ${CMAKE_BINARY_DIR}/include/coreneuron/utils)
+configure_file(${MODFUNC_PERL_SCRIPT} ${CMAKE_BINARY_DIR}/share/coreneuron/mod_func.c.pl COPYONLY)
+configure_file(${KINDERIV_PYTHON_SCRIPT} ${CMAKE_BINARY_DIR}/share/coreneuron/kinderiv.py COPYONLY)
+configure_file(${DIMPLIC_CODE_FILE} ${CMAKE_BINARY_DIR}/share/coreneuron/dimplic.cpp COPYONLY)
+configure_file(${ENGINEMECH_CODE_FILE} ${CMAKE_BINARY_DIR}/share/coreneuron/enginemech.cpp COPYONLY)
+
+# NOTE: this doesn't get re-copied if source changes
```
Comment (ohm314) [inline coreneuron/CMakeLists.txt]: should this not be addressed then? maybe a `custom_target` or `custom_command` with `${CMAKE_COMMAND} -E copy_if_different` would be better?

Comment (ferdonline) [reply coreneuron/CMakeLists.txt]: Meybe lets keep that for yet another time. It's these cases that we have to write 20loc with a loop that might under certain conditions affect someone. That someone would have to be working on nrnivmodl-core, which is probably me or you 🤣

**Comment by ferdonline** [comment]
retest this please

**Comment by ferdonline** [comment]
@jorblancoa I believe this was required to build coreneuron+nmodl but I need a memory refresh.
Anyway what was blocking it?

File: coreneuron/CMakeLists.txt (hunk 1 above)
Comment (ferdonline) [reply coreneuron/CMakeLists.txt]: Being addressed now by doing configure_file on all of them

**Comment by ferdonline** [comment]
CI is down 😢

**Comment by ferdonline** [comment]
@pramodk how to fix the CI?

**Comment by ferdonline** [comment]
@pramodk please review the last patch which is supposed to bring CI to run again

**Comment by pramodk** [comment]
all green! going to merge this now!

**Pull request merged**